app/src/main/java/org/sopt/dosopttemplate/presentation/auth/SignUpActivity.kt

```diff
@@ -19,23 +18,54 @@ class SignUpActivity : AppCompatActivity() {
         super.onCreate(savedInstanceState)
         setContentView(binding.root)
 
-        binding.btnSignupSignup.setOnClickListener {
-            val signUpUserId = binding.etSignupId.text.toString()
-            val signUpUserPw = binding.etSignupPw.text.toString()
-            val signUpUserNickname = binding.etSignupNickname.text.toString()
-            // val signUpUserAge = binding.etSignupAge.text.toString()
+        binding.lifecycleOwner = this
+        binding.authViewModel = signUpViewModel
+
+        observeValid()
+        clickSignUpBtn()
+    }
+
+    private fun observeValid() {
+        signUpViewModel.idFlag.observe(this) { idFlag ->
+            if (idFlag) {
+                binding.telSignupId.error = null
+            } else {
+                binding.telSignupId.error = getString(R.string.id_layout_title)
```

Comment:
```suggestion
            binding.telSignupId.error =  if (idFlag) null else getString(R.string.id_layout_title)
```
요런 식으로 더 깔꼼하게 바꿀수도 있을듯~~

Reply:
수정했어요!
그리고 위에 읽어봤는데 제가 하던 고민이랑 같은 거 같아여... 좋은 정보 감삼다 ㅎㅎ

Reply:
오호,,,신기하네용